service/controller/resource/statuscondition/create_test.go

```diff
@@ -317,3 +317,370 @@ func newCommonClusterObjectFunc(provider string) func() infrastructurev1alpha2.C
 		return new(infrastructurev1alpha2.AWSCluster)
 	}
 }
+
+func TestComputeCreatingCondition(t *testing.T) {
+	testCases := []struct {
+		name string
+
+		status         infrastructurev1alpha2.CommonClusterStatus
+		expectedResult bool
+	}{
+		// There are no previous versions or conditions in the status
+		{
+			name: "case 0",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{},
+				Versions:   []infrastructurev1alpha2.CommonClusterStatusVersion{},
+			},
+			expectedResult: true,
+		},
+		// There are previous versions in the status
+		{
+			name: "case 1",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(5, "8.7.6"),
+				},
+			},
+			expectedResult: false,
+		},
+		// There are previous conditions in the status
+		{
+			name: "case 2",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{},
+			},
+			expectedResult: false,
+		},
```

Comment:
I would add more cases for versions and conditions being present. And then also multiples of that. Too much coverage is better than too little. Especially in projects with multiple and potentially changing contributors over time.